### PR TITLE
feat: add locations endpoint with JWT integration

### PR DESCRIPTION
**Pull request by jasonwalsh**
### Description

This pull request adds a `/v1/square/locations` endpoint. All requests to this endpoint require an `Authorization` HTTP header with the encoded JWT token returned from the `/v1/square/redirect` endpoint.

### Testing

Currently, integration testing requires deploying the application to AWS. We should consider using the official [`dynamodb-local`](https://hub.docker.com/r/amazon/dynamodb-local) Docker image to run integration tests locally.

- [ ] Run the Ansible playbook:
  `ansible-playbook -i localhost, playbook.yml -e '{"parameter_overrides": ["ClientToken=...", "ClientSecret=...", "JwtKey=..."]}'`
- [ ] Visit https://api.hopdrops.io/v1/square/oauth
- [ ] `curl -H 'Authorization: Bearer JWT_TOKEN' https://api.hopdrops.io/v1/square/locations`

The response should be similar to the following:

```json
[
    {
        "address": {
            "address_line_1": "...",
            "administrative_district_level_1": "...",
            "country": "US",
            "locality": "...",
            "postal_code": "..."
        },
        "business_hours": {},
        "capabilities": [
            "CREDIT_CARD_PROCESSING"
        ],
        "coordinates": {
            "latitude": 0,
            "longitude": -0
        },
        "country": "US",
        "created_at": "2018-06-02T14:38:11Z",
        "currency": "USD",
        "id": "04ZQZNFCH3K00",
        "language_code": "en-US",
        "merchant_id": "...",
        "name": "...",
        "status": "ACTIVE",
        "timezone": "America/New_York",
        "type": "PHYSICAL"
    }
]
```

### CODE REVIEW

**Comment by AmillerSC** [comment]
If it isn't obvious @brh55 - you need to login with a personal Square merchant account to fully test this. I think @jasonwalsh and myself have created one each with multiple locations.

I ran this myself and it worked with similar output. We will add expiry/secret rotation in later releases.

File: functions/locations.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from http import HTTPStatus
+from json import dumps
+
+from square.client import Client
+
+try:
+    from decorators import auth
```
Comment (AmillerSC) [inline functions/locations.py]: can you comment this? I know there was confusion

File: functions/decorators.py [hunk 2]
```diff
@@ -30,7 +31,7 @@ def wrapper(event, context):
                 'statusCode': 400
             }
 
-        event['headers']['x-access-token'] = response['Item']['access_token']
+        event['headers']['X-Access-Token'] = response['Item']['access_token']
```
Comment (brh55) [inline functions/decorators.py]: good catch

**Comment by brh55** [comment]
Create an issue to pass token to a `/setup#token=JWT_HERE`, I say we use a hash param as it doesn't get passed to the server to prevent JWTs from showing up in our logs. 

https://stackoverflow.com/questions/15238391/hash-params-vs-url-params-when-to-use-which

**Pull request merged**